### ISSUE TITLE
🐛 Bug Report — Runtime APIs — Invalid URL port component (causes error with yoga-graphql)

### ISSUE THREAD

**Issue by rmtmckenzie** [cloudflare/workerd#1880]
I'm using workers to host a graphql server using Yoga-GraphQL. It has been working but after updating the workers runtime, I've been getting an error when non-graphql pages (i.e. "/"): `Uncaught TypeError: Invalid URL port component`

This appears to be as a result of a URLPattern being used to test against the "/graphql" pathname. The following can reproduce the error:

```
const pattern = new URLPattern({pathname: "/graphql"});
const url = new URL("http://localhost:53000/");

const result = pattern.test(url);
```

In chrome and using the `urlpattern-polyfill` npm package, the call to test results in `false`, but in the workerd runtime it is currently resulting in the aforementioned error.

I've traced that error back to the [source](https://github.com/cloudflare/workerd/blob/bfcfe9a7b2518a93d6c6d176be2f45d27e7c0913/src/workerd/jsg/url.c%2B%2B#L2232), which appears to be called from URLPattern:test.

I have a suspicion the issue is that the URLPattern is creating a URL with "*" for the port.

```
// console.log of the URLPattern doing the testing
URLPattern {
  hash: '*',
  search: '*',
  pathname: '/graphql',
  port: '*',
  hostname: '*',
  password: '*',
  username: '*',
  protocol: '*'
} 

// console.log of the URL being tested
URL {
  searchParams: URLSearchParams(0) {},
  hash: '',
  search: '',
  pathname: '/',
  port: '53000',
  hostname: 'localhost',
  host: 'localhost:53000',
  password: '',
  username: '',
  protocol: 'http:',
  href: 'http://localhost:53000/',
  origin: 'http://localhost:53000'
}```

**Comment by rmtmckenzie** [comment]
For anyone running into this, there's a fairly simple temporary workaround:

First install the urlpattern-polyfill package
```
> npm i urlpattern-polyfill
```

Then in your `index.ts` or wherever you're instantiating your yoga server with `createYoga`, simply pass in the alternate URLPattern to the `fetchAPI` option:

```
export const yoga = createYoga({
  fetchAPI: {
    URLPattern: URLPattern,
  },
  ....
}
```

**Comment by HippocampusGirl** [comment]
I just ran into a similar issue :-) 

Looking at the `workerd` unit tests for `URLPattern`, it seems like their implementation only works for string inputs to `URLPattern.test`, not for `URL` objects (which seem to be the problem here).
https://github.com/cloudflare/workerd/blob/bfcfe9a7b2518a93d6c6d176be2f45d27e7c0913/src/workerd/api/tests/url-test.js#L9363C5-L9364C51

**Comment by jasnell** [comment]
The issue was a small typo (a `>=` that should have just been a `>`) in the port component verification in the URLPattern impl. Fixed in #1881 ...

**Comment by rmtmckenzie** [comment]
Awesome thanks for the quick fix
